src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/ProtobufSerializerHelper.cs

```diff
@@ -0,0 +1,233 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace OpenTelemetry.Exporter.Geneva;
+
+internal static class ProtobufSerializerHelper
+{
+    private const int Fixed64Size = 8;
+
+    internal static Encoding Utf8Encoding => Encoding.UTF8;
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteStringTag(byte[] buffer, ref int currentPosition, string value, int fieldNumber)
+    {
+        int stringSize = Utf8Encoding.GetByteCount(value);
+
+        WriteTag(buffer, ref currentPosition, fieldNumber, WireFormat.WireType.LengthDelimited);
+
+        WriteLength(buffer, ref currentPosition, stringSize);
+
+        _ = Encoding.UTF8.GetBytes(value, 0, value.Length, buffer, currentPosition);
+
+        currentPosition += stringSize;
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteEnumWithTag(byte[] buffer, ref int currentPosition, int fieldNumber, int value)
+    {
+        WriteTag(buffer, ref currentPosition, fieldNumber, WireFormat.WireType.Varint);
+
+        // Assuming 1 byte which matches the intended use.
+        // Otherwise, need to first calculate the bytes needed.
+        WriteRawByte(buffer, ref currentPosition, (byte)value);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteBoolWithTag(byte[] buffer, ref int currentPosition, int fieldNumber, bool value)
+    {
+        WriteTag(buffer, ref currentPosition, fieldNumber, WireFormat.WireType.Varint);
+        WriteRawByte(buffer, ref currentPosition, value ? (byte)1 : (byte)0);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteFixed64WithTag(byte[] buffer, ref int currentPosition, int fieldNumber, ulong value)
+    {
+        WriteTag(buffer, ref currentPosition, fieldNumber, WireFormat.WireType.Fixed64);
+        WriteRawLittleEndian64(buffer, ref currentPosition, value);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static void WriteInt64WithTag(byte[] buffer, ref int currentPosition, int fieldNumber, ulong value)
```

Comment:
Use the same convention of having the proto type name in the method name.

```suggestion
    internal static void WriteFixed64WithTag(byte[] buffer, ref int currentPosition, int fieldNumber, ulong value)
```

Reply:
This is different than fixed64. It used varint https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/9865323c25f0a55f52fae2221faef4b4338d990e/src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/ProtobufSerializerHelper.cs#L59